packages/eslint-plugin/src/rules/consistent-indexed-object-style.ts

```diff
@@ -269,3 +267,99 @@ function findParentDeclaration(
   }
   return undefined;
 }
+
+function getMembers(
+  node: TSESTree.TSInterfaceDeclaration | TSESTree.TSTypeLiteral,
+): TSESTree.TypeElement[] {
+  if (node.type === AST_NODE_TYPES.TSInterfaceDeclaration) {
+    return node.body.body;
+  }
+
+  return node.members;
+}
+
+function isDeeplyReferencingType(
+  node: TSESTree.Node,
+  superVar: ScopeVariable,
+  visited: Set<TSESTree.Node>,
+): boolean {
+  if (visited.has(node)) {
+    // something on the chain is circular but it's not the reference being checked
+    return false;
+  }
+
+  visited.add(node);
+
+  switch (node.type) {
```

Comment:
I went over the possibilities and I think I got them all, though it's possible I missed some 🙈

Reply:
This is pretty thorough and for a not-very-common case. I'm never surprised by edge cases popping up in _"check all kinds of nodes/types"_ logic, but would be surprised if many came out of this.